src/pages/[guild]/index.tsx

```diff
@@ -182,8 +189,22 @@ const GuildPage = (): JSX.Element => {
     if (isAdmin && !contacts?.length && !isLoading) showAddContactInfoToast()
   }, [isAdmin, contacts, isLoading])
 
+  useEffect(() => {
+    const handleRouteChange = (url) => {
+      if (url == `/${urlName}` || url == `/${id}`) return
+      toast.close(CONTACT_TOAST_ID)
+    }
+
+    router.events.on("routeChangeStart", handleRouteChange)
+    return () => {
+      router.events.off("routeChangeStart", handleRouteChange)
+    }
+  }, [router.events])
+
   const showAddContactInfoToast = () => {
+    if (toast.isActive(CONTACT_TOAST_ID)) return
```

Comment:
interestingly, toastIdRef.current was not working, I needed to check the toast by id
however, couldn't we remove this check as we close the toast on navigation anyways?

also, can we manage closing the toast without using the useToast hooks close function?

Reply:
Ah, I guess it didn't work because the page got unmounted on navigation, so `toastIdRef.current` was undefined. But yeah, we might just remove this check actually!